evalml/pipelines/utils.py

```diff
@@ -1348,3 +1349,109 @@ def rows_of_interest(
 
     preds_value_proba = preds_value_proba[preds_value_proba <= epsilon]
     return preds_value_proba.index.tolist()
+
+
+def unstack_multiseries(X, y, series_id, time_index, keep_time_in_index=True):
+    """Converts multiseries data with one series_id column and one target column to one target column per series id.
+
+    Args:
+        X (pd.DataFrame): Data of shape [n_samples, n_features].
+        y (pd.Series): Target data.
+        series_id (str): The column which identifies which series each row belongs to.
+        time_index (str): Specifies the name of the column in X that provides the datetime objects.
+        keep_time_in_index (bool): Whether to maintain the time index as the index of the returned dataframes. Defaults to True.
+            If set to false, will discard the time index information entirely.
+
+    Returns:
+        pd.DataFrame, pd.DataFrame: The unstacked X and y data.
+    """
+    # Save the target name so we can re-separate it
+    target_name = y.name
```

Comment:
I'm curious what you mean by "bugs out" - what situation would that be? Having `target_name` be a required argument would mean either the user would have to manually pass in the name of their target column and ensure that matches their data, or we would have to pull out the name ourselves with the same method here, just in a different spot. Thoughts?

Reply:
Just if it ever works as not expected 😄 

I think that makes sense to me since `time_index` behaves similarly (`series_id` could in the future as well) but really a small nit!